File: cpp/easy/0338_counting_bits.cpp
// https://leetcode.com/problems/counting-bits/description/
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> v(n + 1, 0);
        int x, count;

        for(int i = 0; i <= n; i++){
            x = i;
            count = 0;
            while(x != 0){
                x = x & (x - 1);
                count++;
            }
            v[i] = count;
        }
        return v;
    }
};

File: python/easy/0338_counting_bits.py
# https://leetcode.com/problems/counting-bits/description/
class Solution:
    def countBits(self, n: int) -> List[int]:
        res = [0] * (n + 1)
        res[0] = 0
        for i in range(1, n + 1):
            res[i] = res[i >> 1] + (i & 1)
        return res
